src/Microsoft.OpenApi/Models/OpenApiReference.cs

```diff
@@ -201,7 +201,7 @@
                     return ExternalResource + "#" + Id;
                 }
 
-                return ExternalResource + "#/components/" + Type.GetDisplayName() + "/"+ Id;
+                return ExternalResource + "#/components/" + Type.Value.GetDisplayName() + "/"+ Id;
```

Comment:
This warning is interesting. The code comments on the property say:

```
/// <remarks>This must be present if <see cref="ExternalResource"/> is not present.</remarks>
public ReferenceType? Type { get; set; }
```

But they don't say anything about the inverse: if `ExternalResource` is present, can `Type` be null? I assume in that case, we'd want to avoid this codepath altogether and just return `ExternalResource`?

Reply:
If we could bind to nullable type instead it would probably solve this error

Reply:
We could, but this feels like a logical gap to me. Given the lack of null-checking in the current codepath, it strikes me that there is a break in the code if ExternalResource = true and Type = null?

Reply:
External references resolution has been a major issue with this version. We've revamped it entirely for the upcoming v2.
I'd be ok with leaving it as is and consider it legacy code as long as we're not breaking things further. 
I'd also like @maggiekimani1 input on this one

Reply:
@MaggieKimani1 Any thoughts on this? Would love to move this PR forward and I think this is the last outstanding item 😄

Reply:
I agree with @baywet on this one. We've optimized both local and external reference resolution in the next major version so we can leave this as is pending further investigation.

Reply:
OK -- I wasn't sure if "leave as is" meant that we want to keep the CodeQL warning around. I pused https://github.com/microsoft/OpenAPI.NET/pull/1717/commits/212840898f980b7b4214fcd6a329012693fde1a7 with a guard for the value to remove the warning.

If someone was running into this before, they would've encountered a NRE in the codepath. Now, we'll produce just a reference to the external resource and avoid mapping to a particular reference ID.

Let me know if this is not the behavior we want to encode.